KUBEVIRTCI_LOCAL_TESTING.md

```diff
@@ -1,40 +1,90 @@
-                                                                                          
-### start cluster and test                                                                
-                                                                                          
-```bash                                                                                   
-cd .../kubevirt                                                                           
-                                                                                          
-# spin up cluster                                                                         
-make cluster-up                                                                           
-                                                                                          
-# deploy latest kubevirt changes to cluster                                               
-make cluster-sync                                                                         
-                                                                                          
-# start tests, either                                                                     
-make functest                                                                             
-                                                                                          
-# or use ginkgo focus                                                                     
+```
+
+Note:
+In case you updated gocli and need to test it locally as well, export additionally:
+```bash
+export KUBEVIRTCI_GOCLI_CONTAINER=quay.io/kubevirtci/gocli:latest                                     
+```
+
+### start cluster
+
+```bash        
+# switch back to kubevirtci folder
+cd $KUBEVIRTCI_DIR
+
+export KUBEVIRT_PROVIDER=k8s-1.19
+export KUBECONFIG=$(./cluster-up/kubeconfig.sh)
+export KUBEVIRT_NUM_NODES=2
+
+# spin up cluster
+make cluster-up
+```
+
+## kubevirt: testing kubevirt locally with a freshly provisioned cluster
+
+After making changes to a kubevirtci provider, it's recommended to test it locally including kubevirt e2e tests before publishing it.
+
+With the changes in place you can execute locally [`make functest`](https://github.com/kubevirt/kubevirt/blob/master/docs/getting-started.md#testing) against a cluster with kubevirt that was provisioned using `kubevirtci`.
+
+`$KUBEVIRT_DIR` assumed to be your kubevirt path.
+
+Steps:
+
+### sync cluster-up folder
+
+```bash
+# sync cluster-up folder if it has changed.
+rsync -av $KUBEVIRTCI_DIR/cluster-up $KUBEVIRT_DIR/cluster-up
+```
+
+### start cluster
+Note: make sure to `make cluster-down` any previous clusters, if any.
+
+```bash
+# switch directory to kubevirt folder
+cd $KUBEVIRT_DIR
+
+export KUBEVIRT_PROVIDER=k8s-1.19
+export KUBECONFIG=$(./cluster-up/kubeconfig.sh)
+export KUBEVIRT_NUM_NODES=2
+
+# spin up cluster
+make cluster-up
```

Comment:
why is this needed? we already have a cluster ready from the former (kubevirtci-only) step

Reply:
Theoretical its might not mandatory
but then you need to copy the `_ci-configs/` to `kubevirt` folder, because some portions of kubevirt e2e tests
are looking hardcoded in $KUBEVIRT_PROVIDER/_ci_configs

best also to export those two again
```
export KUBEVIRT_PROVIDER=k8s-1.19
export KUBECONFIG=$(./cluster-up/kubeconfig.sh)
```

I find it better to recreate a cluster, within the kubevirt folder, and test it end 2 end
instead doing the two hacks above

this way, it also make this step independent of the previous `cluster-up` step of kubevirtci
wdyt?

Reply:
> wdyt?

I don't no what is `ci_config` and I don't understand anything. How is the new cluster different from the old? As I see it, kubevirtci's `make cluster-up` + rsync + kubevirt's `make cluster-sync` should be enough. If you have to take the cluster down and use kubevirt's `make cluster-up` please explain what is the difference between the two `cluster-up`s.

Reply:
You don't need to cluster-down it, but you must copy `_ci_configs` folder to kubevirt,
else you will not be able to `cluster-sync` even,
this folder contains for example information about the ports of the cluster IIRC.
I just tried it.

I will add the `_ci_configs` to the rsync snippet and update the docs accordingly.